src/scanner.cc

```diff
@@ -146,7 +146,8 @@ struct Scanner {
 
     for (;;) {
       if (position_in_word == heredoc.word.size()) {
-        if (lexer->lookahead == ';' || lexer->lookahead == '\n' || lexer->lookahead == '\r') {
+        // , and ) is needed to support heredoc in function arguments
+        if (lexer->lookahead == ';' || lexer->lookahead == ',' || lexer->lookahead == ')' || lexer->lookahead == '\n' || lexer->lookahead == '\r') {
```

Comment:
After thinking a bit more about this, I realize that while this works in the ideal case, we might run into problems if e.g. the user places a space before the comma or closing parenthesis.

Should I just add a special case checking all whitespace until the next char or is there some idiomatic tree-sitter way of handling this?

Reply:
Yeah, there's no built-in facility for this, so I think you have to manually skip over whitespace.